plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/DeploymentResourceMapper.java

```diff
@@ -129,24 +124,12 @@ private void load(String kind) {
 
 	private void buildDeployments() {
 		if (state.compareAndSet(State.UNINITIALIZED, State.LOADING)) {
-			for (String kind : WatchManager.KINDS) {
-				load(kind);
-			}
 			WatchManager.getInstance().stopWatch(project);
+			ConnectionsRegistrySingleton.getInstance().removeListener(connectionListener);
```

Comment:
One use case that I see is that while build is going user calls refresh. Even though there may be not much resources, the time of refresh may be not negligible because each kind is retrieved with a separate remote request. Taken that build goes for several minutes, and refresh for several seconds, we get probability that the build status changes after it is loaded but before refresh is completed about several per cents. So, some logic to prevent that should be implemented, the one you have described sounds good.